aiida/engine/daemon/client.py

```diff
@@ -467,8 +515,9 @@ def stop_daemon(self, wait: bool = True, timeout: int = 5) -> JsonDictType:
         :return: The client call response.
         :raises DaemonException: If ``is_daemon_running`` returns ``True`` after the ``timeout`` has passed.
         """
-        command = {'command': 'quit', 'properties': {'waiting': wait}}
+        self._clean_potentially_stale_pid_file()
```

Comment:
> Because there would certainly be an upside in the shared FS scenario...

Well the problem is still there if people run `verdi daemon start` from the other machine right? It doesn't really solve it.

But I guess it is ok as long as the logic for starting the daemon never skips starting the daemon if a PID file exists. Because that would cause a "dead-lock" scenario in case of an actual stale PID. But that is currently not the case, so I guess we can have the check just in the daemon start method.

Reply:
> Well the problem is still there if people run verdi daemon start from the other machine right? It doesn't really solve it.

Correct, it just reduces the number of commands that cause removal of the PID file.

By the way, I am just suggesting to be conservative about deleting the file; I think `verdi daemon stop/restart` about the existence of a stale PID file (and if I understand correctly, the existence of a stale PID file actually means that these two commands will fail).